scipy/cluster/hierarchy.py

```diff
@@ -3392,7 +3392,8 @@ def _get_leaves_color_list(R):
                                           R['dcoord'],
                                           R['color_list']):
         for (xi, yi) in zip(link_x, link_y):
-            if yi == 0.0:  # if yi is 0.0, the point is a leaf
+            if yi == 0.0 and xi % 5 == 0.0 and xi % 2 == 1:  
```

Comment:
I think this is more readable, what do you think?
```suggestion
            if yi == 0.0 and (xi % 5 == 0 and xi % 2 == 1):  
```